Sources/Utils/Log.cpp

```diff
@@ -66,9 +81,28 @@ uint32_t GetLogLevel() { return sLogLevel; }
 
 void SetLogLevel(uint32_t level) { sLogLevel = level; }
 
-void SetLogColorsEnabled(bool enabled) { sColorsEnabled = enabled; }
+const char *GetLogOutputFilename() { return sOutputFilename; }
+
+void SetLogOutputFilename(const char *filename) {
+  FILE *stream = fopen(filename, "a");
+  if (stream == nullptr) {
+    DS2LOG(Error, "unable to open %s for writing: %s", filename,
+           strerror(errno));
+  } else {
+#if defined(OS_POSIX)
+    // When ds2 is spawned by an app (e.g.: on Android), it will run with the
+    // app's user/group ID, and will create its log file owned by the app. By
+    // default, the permissions will be 0600 (rw-------) which makes us
+    // unable to get the log files. chmod() them to be able to access them.
```

Comment:
Why does this apply to all POSIX OSes instead of just android? Would it happen on Linux or Tizen or any other POSIX platform we support? I know this isn't your code since you got it from `main.cpp`, but making it more specific couldn't hurt. :)

Reply:
This behavior depends on what the process' `umask` is. By default, it'll be `022` which means the log file will be created with mode `755` but on Android the umask happens to be `077` sometimes. Always forcing the file to be readable makes things a little bit easier.

That being said, if `umask` is *not* `022` on Linux, it's probably set by the user for a reason, so making this `fchmod()` only on Android could be a better behavior.